scipy/stats/_stats_py.py

```diff
@@ -900,13 +870,10 @@ def tsem(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
     1.1547005383792515
 
     """
-    a = np.asarray(a).ravel()
-    if limits is None:
-        return a.std(ddof=ddof) / np.sqrt(a.size)
-
-    am = _mask_to_limits(a, limits, inclusive)
-    sd = np.sqrt(np.ma.var(am, ddof=ddof, axis=axis))
-    return sd / np.sqrt(am.count())
+    a = _put_nan_to_limits(a, limits, inclusive)
+    sd = np.sqrt(np.nanvar(a, ddof=ddof, axis=axis))
+    n_samples = (~np.isnan(a)).sum(axis=axis)
```

Comment:
I try to avoid the unnecessary ambiguity with the word "sample".
```suggestion
    n_obs = (~np.isnan(a)).sum(axis=axis)
```
I don't think it should be allowed to mean both "observation" and "collection of observations".